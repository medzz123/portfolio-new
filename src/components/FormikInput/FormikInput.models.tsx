import { TextInputProps } from '@components/TextInput/TextInput.models';

export interface FormikInputProps extends TextInputProps {
  /**
   * Name will be used to lookup the values tree in Formik
   * . notation such as address.postcode works
   */
  name: string;
}
