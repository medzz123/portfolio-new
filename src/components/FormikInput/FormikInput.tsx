import TextInput from '@components/TextInput';
import { useField } from 'formik';
import React, { FunctionComponent } from 'react';

import { FormikInputProps } from './FormikInput.models';

const FormikInput: FunctionComponent<FormikInputProps> = (props) => {
  const { name, ...rest } = props;
  const [field, meta, helpers] = useField(name);

  const error = meta.touched && Boolean(meta.error);

  return (
    <TextInput
      error={error}
      errorMessage={meta.error}
      placeholder={rest.placeholder}
      value={field.value}
      onChange={(event) => {
        helpers.setValue(event.target.value);
      }}
      onBlur={() => {
        helpers.setTouched(true);
      }}
      name={name}
      {...rest}
    />
  );
};

export default FormikInput;
