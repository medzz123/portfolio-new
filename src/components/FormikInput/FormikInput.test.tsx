import { render } from '@test/library';
import { Formik } from 'formik';
import React from 'react';

import FormikInput from './FormikInput';

test('FormikInput â†’ Renders Correctly', () => {
  const { container } = render(
    <Formik<{ test: string }>
      initialValues={{ test: 'hu' }}
      onSubmit={() => {
        jest.fn();
      }}
    >
      <FormikInput name="test" />
    </Formik>
  );

  expect(container).toMatchSnapshot();
});
