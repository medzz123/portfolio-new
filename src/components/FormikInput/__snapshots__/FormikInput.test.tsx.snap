// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikInput â†’ Renders Correctly 1`] = `
<div>
  <input
    class="sxp0t8a sxp0t8a3bov9--color-normal"
    data-testid="textInput"
    name="test"
    value="hu"
  />
</div>
`;
